lib/configure.js

```diff
@@ -360,3 +278,97 @@ function configure (gyp, argv, callback) {
   }
 
 }
+
+function findPython (python, callback) {
+  checkPython()
+
+  // Check if Python is in the $PATH
+  function checkPython () {
+    log.verbose('check python', 'checking for Python executable "%s" in the PATH', python)
+    which(python, function (err, execPath) {
+      if (err) {
+        log.verbose('`which` failed', python, err)
+        if (python === 'python2') {
+          python = 'python'
+          return checkPython()
+        }
+        if (win) {
+          guessPython()
+        } else {
+          failNoPython()
```

Comment:
After the refactoring, `failNoPython` directly calls the `configure`'s `callback` and brings the execution to end. But ideally it should have called `findPython`'s `callback` and that should have propagated the error back to the caller of `configure`. This will still work but looks like bad design to me. Please point out if I am missing something obvious.